catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitor.java

```diff
@@ -326,20 +348,43 @@ private String getBlackListAsRegex() {
      automatically start after a system shutdown.
   */
   private void attemptAddRoutes() {
-    LOGGER.trace(
-        "Attempting to add routes for content directory monitor watching {}", monitoredDirectory);
-    try {
-      RouteBuilder routeBuilder = createRouteBuilder();
-      verifyCamelComponentIsAvailable("content");
-      verifyCamelComponentIsAvailable("catalog");
-      camelContext.addRoutes(routeBuilder);
-      setRouteCollection(routeBuilder);
-    } catch (Exception e) {
-      LOGGER.info("Unable to configure Camel route.", e);
-      INGEST_LOGGER.warn("Unable to configure Camel route.", e);
-    } finally {
-      if (LOGGER.isDebugEnabled()) {
-        dumpCamelContext("after attemptAddRoutes()");
+    Set<String> inputTransformerIds = InputTransformerIds.getTransformerIds();
+    synchronized (initState) {
+      try {
+        cdmServiceWaiter.waitFor(inputTransformerIds);
+      } catch (InterruptedException e) {
+        LOGGER.debug(
+            "Interrupted while waiting for input transformers. CDM route will not be started.", e);
```

Comment:
Good point. I'd imagine the configuration will still be saved since this is all done on a separate thread. Not sure what to do about that.

Reply:
This is making me re-think if MSF threads block blueprint or not, I will check it out. I had assumed it does because we were always spinning off initialization in another thread.